sim/hunter/items.go

```diff
@@ -148,6 +153,124 @@ func init() {
 		hunter := agent.(HunterAgent).GetHunter()
 		hunter.newBloodlashProcItem(75, 446723)
 	})
+
+	// https://www.wowhead.com/classic/item=228432/whistle-of-the-beast
+	// Use: Your pet's next attack is guaranteed to critically strike if that attack is capable of striking critically. (1 Min Cooldown)
+	core.NewItemEffect(WhistleOfTheBeast, func(agent core.Agent) {
+		hunter := agent.(HunterAgent).GetHunter()
+
+		if hunter.pet != nil {
+			hunter.pet.PseudoStats.DamageDealtMultiplier *= 1.03
+			hunter.pet.MultiplyStat(stats.Health, 1.03)
+			hunter.pet.MultiplyStat(stats.Armor, 1.10)
+			hunter.pet.AddStat(stats.MeleeCrit, 2*core.CritRatingPerCritChance)
+			hunter.pet.AddStat(stats.SpellCrit, 2*core.SpellCritRatingPerCritChance)
+		}
+
+		actionID := core.ActionID{ItemID: WhistleOfTheBeast}
+
+		trackingAura := hunter.GetOrRegisterAura(core.Aura{
+			Label:    "Whistle of the Beast Hunter",
+			ActionID: actionID,
+			Duration: core.NeverExpires,
+		})
+
+		aura := hunter.pet.GetOrRegisterAura(core.Aura{
+			Label:    "Whistle of the Beast",
+			ActionID: actionID,
+			Duration: core.NeverExpires,
+
+			OnGain: func(aura *core.Aura, sim *core.Simulation) {
+				if hunter.pet.focusDump != nil {
+					hunter.pet.focusDump.BonusCritRating += 100
+				}
+				if hunter.pet.specialAbility != nil {
+					hunter.pet.specialAbility.BonusCritRating += 100
+				}
+				trackingAura.Activate(sim)
+			},
+			OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+				if hunter.pet.focusDump != nil {
+					hunter.pet.focusDump.BonusCritRating -= 100
+				}
+				if hunter.pet.specialAbility != nil {
+					hunter.pet.specialAbility.BonusCritRating -= 100
+				}
+				trackingAura.Deactivate(sim)
+			},
+			OnSpellHitDealt: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if spell == hunter.pet.focusDump || spell == hunter.pet.specialAbility {
+					aura.Deactivate(sim)
+				}
+			},
+		})
+
+		spell := hunter.GetOrRegisterSpell(core.SpellConfig{
+			ActionID: actionID,
+			Flags:    core.SpellFlagNoOnCastComplete | core.SpellFlagOffensiveEquipment,
+
+			Cast: core.CastConfig{
+				CD: core.Cooldown{
+					Timer:    hunter.NewTimer(),
+					Duration: time.Minute * 1,
+				},
+			},
+			ExtraCastCondition: func(sim *core.Simulation, target *core.Unit) bool {
+				return hunter.pet.IsEnabled()
```

Comment:
Will this potentially error if no pet is selected?

Reply:
Or I wonder do you need this cast condition with the `ShouldActivate` on the major cooldown?

Reply:
Simple fix was just changing the first lines to if return on ==nil